lib/handle-schedule.ts

```diff
@@ -70,6 +73,17 @@ export default async function handleSchedule(): Promise<void> {
   }
 
   for await (const pullRequest of duePullRequests) {
+    if (requireStatusesSuccess) {
+      const { data } = await octokit.rest.repos.getCombinedStatusForRef({
+        ...github.context.repo,
+        ref: pullRequest.ref,
+      });
+      if (data.state !== "success") {
+        core.info(`${pullRequest.html_url} is not ready to be merged yet`);
+        continue;
```

Comment:
yeah, it's a bit confusing for sure.

I was looking into the rest API documentation and don't see a simpler way to get all information.

I'm gonna use your example for this PR.

Reply:
Check the GraphQL schema, you can explore it at https://docs.github.com/en/graphql/overview/explorer, I think there was something introduced recently that would combine statuses from both commit statuses and check runs

Reply:
Will do!

Reply:
I don't anything that combines both commit statuses and check runs.

I used the rest API to fetch commit statuses and check runs instead of using the GraphQL API for better type safety 😉